### PR TITLE
W-12354025: Upgrade to snakeyaml 2.0 to address vulnerability in earrlier  versions

### PR DESCRIPTION
**Pull request by mule-pachanady**
In October of 2022, a [critical flaw](https://nvd.nist.gov/vuln/detail/CVE-2022-1471) was found in the SnakeYAML package, which allowed an attacker to benefit from remote code execution by sending malicious YAML content and this content being deserialized by the constructor. Finally, in February 2023, the SnakeYAML 2.0 release was pushed that resolves this flaw, also referred to as [CVE-2022-1471](https://nvd.nist.gov/vuln/detail/CVE-2022-1471). Let’s break down how this version can help you resolve this critical flaw.

### CODE REVIEW

File: mule-packager/src/test/java/org/mule/tools/api/verifier/ProjectVerifierFactoryTest.java [hunk 1]
```diff
@@ -43,4 +51,13 @@ public void createPreparePackagerTest() {
       }
     }
   }
+
+  @Test
+  @Issue("W-12354025")
+  @Description("Verify that PolicyYamlVerifier validation works after upgrading to snakeyaml 2.0")
+  public void verifyValidation() throws ValidationException {
+    PolicyYamlVerifier policyYamlVerifier =
+        new PolicyYamlVerifier(this.getClass().getResource("/org/mule/tools/api/verifier").getPath(), "custom.policy.test.yaml");
```
Comment (fsgonz) [inline mule-packager/src/test/java/org/mule/tools/api/verifier/ProjectVerifierFactoryTest.java:60]: What is the logic of this test?
Add a comment explaining it.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: mule-packager/src/test/java/org/mule/tools/api/verifier/ProjectVerifierFactoryTest.java (hunk 1 above)
Comment (mule-pachanady) [reply mule-packager/src/test/java/org/mule/tools/api/verifier/ProjectVerifierFactoryTest.java:60]: I had to change PolicyYamlVerifier.validate() as in snakeyaml 2.0 the constructor of Yaml class changed. This is a test to validate we can correctly validate the yaml after this change and it is still able to read fields from the Yaml

**Review by fsgonz — DISMISSED** [review]

**Review by fsgonz — APPROVED** [review]

**Comment by mule-pachanady** [comment]
@mulesoft/team-munit Can you please take a look and approve if it looks good, thanks

**Comment by mule-pachanady** [comment]
@mpereyra-salesforce can you please take a look at this please

**Review by rodriguez-jorge — COMMENTED** [review]
Minor comment

File: mule-packager/pom.xml [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
         <coverageLineLimit>0.74</coverageLineLimit>
         <coverageBranchLimit>0.69</coverageBranchLimit>
+        <allure.version>2.8.1</allure.version>
```
[the remark this answers is not in the document]
Comment (mule-pachanady) [reply mule-packager/pom.xml]: I have moved allure.version that I added to parent,, however coverageLineLimit, coverageBranchLimit were overridden here may be for some reason befoe, i dont want to touch it

**Review by rodriguez-jorge — APPROVED** [review]
LGTM

**Pull request merged**